GameLibrary/Unit.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+
+namespace GameLibrary
+{
+    public abstract class Unit
+    {
+        private double _healthPoints;
+        private double _attackDamage;
+
+        public double Health => _healthPoints;  
+        public double CurrentAttackPower => _attackDamage;
+
+        protected Unit(double health, double attack)
+        {
+            this._healthPoints = health;
+            this._attackDamage = attack;
+        }
+
+        protected abstract double GetAttackRate();
+
+        protected abstract double Defence(double atackRate);
+             
+        public void Attack(Unit unit)
+        {
+            double currentAttack = GetAttackRate();
+            this._attackDamage = currentAttack;
```

Comment:
We talked that it's bad idea to rewrite `_attackDamage` and you should either rename or remove it